build-coordinator/src/main/java/org/jboss/pnc/core/builder/BuildCoordinator.java

```diff
@@ -105,8 +105,7 @@ public BuildTask build(BuildConfiguration buildConfiguration, User userTriggered
         BuildSetTask buildSetTask = new BuildSetTask(
                 this,
                 buildConfigurationSet,
-                BuildExecutionType.STANDALONE_BUILD,
-                () -> datastore.getNextBuildConfigSetRecordId());
+                BuildExecutionType.STANDALONE_BUILD, userTriggeredBuild, datastore.getNextBuildConfigSetRecordId());
```

Comment:
This way `BuildSetTask` doesn't know anything about this supplier (loose coupling). BTW it was eager also before (id was extracted during `BuildSetTask` construction). This is why it didn't make any sense to me...